### PR TITLE
Add datepicker

### PR DESCRIPTION
**Pull request by JaKXz**

### CODE REVIEW

File: demo.html [hunk 1]
```diff
@@ -2,11 +2,21 @@
 <head>
```
Comment (jblashill) [inline demo.html]: Can we split this demo into a separate file?

**Comment by jblashill** [comment]
:8ball:

File: src/datepicker.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    replace: true,
+    template: "<input type='text'>",
+    link: function(scope, $el, attrs, ngModel) {
+      var pickerOptions = {
+        dateFormat: attrs.format || 'yy-mm-dd',
+        onClose: function (newValue) {
+          if (newValue === "" && typeof ngModel.$modelValue === 'undefined') {
+            return;
+          }
+          if (newValue !== ngModel.$modelValue.substring(0, 10)) {
```
Comment (JaKXz) [inline src/datepicker.js]: @jblashill just realized this line is probably not great, unless the format is always restricted to 10 characters.

Comment (jblashill) [inline src/datepicker.js]: Let's remove the substring

**Pull request merged**